src/primitives/triangle/scanline_intersections.rs

```diff
@@ -178,51 +177,55 @@ fn generate_lines(
         // triangle (assuming the edge line merging logic is correct). In this case, we need a
         // scanline between the two edge intersections.
         let internal = if has_fill {
-            match (first, second) {
+            match (first.clone(), second.clone()) {
                 // Triangle stroke is non-zero, so the fill line is between the insides of each
                 // stroke.
                 (Some(first), Some(second)) => {
-                    let start_x = first.end.x.min(second.end.x);
-                    let end_x = first.start.x.max(second.start.x);
-
-                    // Line needs to be shrunk by 1px off each end to prevent overdraw.
-                    // This can only happen if there's enough room to do so.
-                    if (end_x - start_x) > 1 {
-                        Some(Line::new(
-                            Point::new(start_x + 1, scanline_y),
-                            Point::new(end_x - 1, scanline_y),
-                        ))
-                    } else {
-                        None
+                    let start_x = first.x.end.min(second.x.end);
+                    let end_x = first.x.start.max(second.x.start);
+
+                    // TODO: check
+                    //  // Line needs to be shrunk by 1px off each end to prevent overdraw.
+                    //  // This can only happen if there's enough room to do so.
+                    //  if (end_x - start_x) > 1 {
```

Comment:
Does this still need checking?

Also, there are two tests in `triangle/styled.rs` that still overdraw: `bounding_box` and `styled_stroke_equals_lines`. Could you remove the other `.set_allow_overdraw(true)` calls in that file as part of this PR? Sorry I didn't pick it up in my thick triangles work. I've opened #489 to track the remaining bugs.

Reply:
> Does this still need checking?

I haven't really looked at the triangle code in detail and only wanted to keep it working while I was refactoring the other code. But I guess as long as no unit test fails it should be safe to remove this note.

> Also, there are two tests in `triangle/styled.rs` that still overdraw: `bounding_box`...

I can't remove it without the test failing and IMO this is outside the scope of this PR.

> ...and `styled_stroke_equals_lines`.

`styled_stroke_equals_lines` only uses `set_allow_overdraw` for the `lines_display` which must be enabled, because drawing single lines will always overdraw on the vertices. But I've removed the unnecessary `set_allow_out_of_bounds_drawing` in that test.